src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -113,6 +114,52 @@ public static long BigMul(int a, int b)
             return ((long)a) * b;
         }
 
+        private static ulong BigMulSoftwareFallback(ulong a, ulong b, out ulong low)
```

Comment:
The usual pattern is to make the software fallback to be a local method inside the main method.

E.g. take a look at https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector256.cs